operator/operator.php

```diff
@@ -53,10 +53,20 @@ public function __construct(\phpbb\config\config $config, \phpbb\db\driver\drive
 	 */
 	public function get_user_categories()
 	{
-		// TODO
-		// 1.get the categories by unserializing the user object data 'collapsible_categories'
-		// 2.if no categories found, call get_cookie_categories()
-		// 3.return the categories or an empty array
+		// Get the categories by json_decode() the user object data 'collapsible_categories'
+		// If no categories found in the database, call get_cookie_categories()
+		// Return the categories array or an empty array if there were not categories found
```

Comment:
We don't need so much commenting, since the interface docblock also explains some of this. We can simplify like:
```php
// Get categories from the user object
if (!empty($this->user->data['collapsible_categories']))
{
	// Data needs to be json decoded and cast as an array
	$collapsible_categories = (array) json_decode($this->user->data['collapsible_categories'], true);
}
else
{
	// The user object had no categories, check for a cookie
	$collapsible_categories = $this->get_cookie_categories();
}
```